integration/integration_test.go

```diff
@@ -313,3 +313,66 @@ func TestDemoteLeader(t *testing.T) {
 	numManager--
 	pollClusterReady(t, cl, numWorker, numManager)
 }
+
+func TestJoinDemoteRace(t *testing.T) {
```

Comment:
I added some debugging output and the call to `cl.RemoveNode` seems to take about 6 seconds.

That's strange. Trying to understand why.